File: model/stats.go
package model

// 统计相关

type DashboardStats struct {
	TotalEvents     int64   `json:"total_events"`
	ActiveEvents    int64   `json:"active_events"`
	TotalTickets    int64   `json:"total_tickets"`
	SoldTickets     int64   `json:"sold_tickets"`
	ReservedTickets int64   `json:"reserved_tickets"`
	TotalRevenue    float64 `json:"total_revenue"`
	TodaySales      int64   `json:"today_sales"`
	TodayRevenue    float64 `json:"today_revenue"`
}

type SalesTrend struct {
	Date    string  `json:"date"`
	Count   int64   `json:"count"`
	Revenue float64 `json:"revenue"`
}

type TicketTypeStats struct {
	TicketTypeID   uint    `json:"ticket_type_id"`
	TicketTypeName string  `json:"ticket_type_name"`
	EventTitle     string  `json:"event_title"`
	SoldCount      int64   `json:"sold_count"`
	Revenue        float64 `json:"revenue"`
}

type ConversionFunnel struct {
	PageViews int64 `json:"page_views"`
	AddToCart int64 `json:"add_to_cart"`
	Reserved  int64 `json:"reserved"`
	Paid      int64 `json:"paid"`
	Used      int64 `json:"used"`
}

File: apps/web/src/api/stats.ts
import request from './request'

export interface DashboardStats {
  total_events: number
  active_events: number
  total_tickets: number
  sold_tickets: number
  reserved_tickets: number
  total_revenue: number
  today_sales: number
  today_revenue: number
}

export interface SalesTrend {
  date: string
  count: number
  revenue: number
}

export interface TicketTypeStats {
  ticket_type_id: number
  ticket_type_name: string
  event_title: string
  sold_count: number
  revenue: number
}

export interface ConversionFunnel {
  page_views: number
  add_to_cart: number
  reserved: number
  paid: number
  used: number
}

export function getDashboardStats() {
  return request.get<DashboardStats>('/admin/stats/dashboard')
}

export function getSalesTrend(days: number = 7) {
  return request.get<{ data: SalesTrend[] }>('/admin/stats/sales-trend', { params: { days } })
}

export function getTicketTypeStats() {
  return request.get<{ data: TicketTypeStats[] }>('/admin/stats/ticket-types')
}

export function getConversionFunnel(eventId: number) {
  return request.get<ConversionFunnel>(`/admin/stats/funnel/${eventId}`)
}
